src/main/java/com/iexec/worker/compute/TeeWorkflowConfiguration.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.worker.compute;
+
+import com.iexec.common.tee.TeeWorkflowSharedConfiguration;
+import com.iexec.worker.docker.DockerService;
+import com.iexec.worker.sms.SmsService;
+import com.iexec.worker.tee.scone.SconeTeeService;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Configuration;
+
+import javax.annotation.PostConstruct;
+
+/**
+ * Configuration of tee workflow. It contains: pre-compute image, pre-compute
+ * heap size, post-compute image, post-compute heap size. Images are downloaded
+ * at initialization.
+ */
+@Slf4j
+@Configuration
+public class TeeWorkflowConfiguration {
+
+    private final SconeTeeService sconeTeeService;
+    private final SmsService smsService;
+    private final DockerService dockerService;
+
+    @Getter
+    private String preComputeImage = "";
+    @Getter
+    private long preComputeHeapSize = 0;
+    @Getter
+    private String postComputeImage = "";
+    @Getter
+    private long postComputeHeapSize = 0;
+
+    public TeeWorkflowConfiguration(
+            SconeTeeService sconeTeeService,
+            SmsService smsService,
+            DockerService dockerService) {
+        this.sconeTeeService = sconeTeeService;
+        this.smsService = smsService;
+        this.dockerService = dockerService;
+    }
+
+    @PostConstruct
+    private void pullPrePostComputeImages() {
```

Comment:
Add note: Workers need to be manually restarted if SMS configuration is updated

Reply:
Can you check what happens if these images are deleted at runtime by admin, do we check for isPulled||pull before executing a pre or post compte run?

Reply:
I will, in each task, check that images on SMS match the local ones and that the local ones exist. Otherwise, we pull the new ones.

Reply:
Not sure we want to re-hit the SMS to check image infos on each run (costly for the SMS), maybe just write it down.
But yes for isImagePresent.

Reply:
The worker needs a way to know when the SMS changes its config. If we don't do it with each task, a session with the new config is created but the pre-compute will be outdated, so it will fail.